### PR TITLE
Consumer exit after fail

### PR DESCRIPTION
**Pull request by PeterWuMC**
@joshuafleck @calo81 

second time lucky!!

so this time i decide to do `exit(1)`, simply re-raising the exception does not help. because bunny rescues any exceptions: https://github.com/ruby-amqp/bunny/blob/master/lib/bunny/consumer_work_pool.rb#L87-L102

### CODE REVIEW

File: spec/lib/rabbit_feed/consumer_connection_spec.rb [hunk 1]
```diff
@@ -79,46 +79,61 @@ module RabbitFeed
...
-        context 'when Airbrake is defined' do
-          after { Object.send(:remove_const, ('Airbrake').to_sym) rescue NameError }
+        context 'when consumer_exit_after_fail is true' do
+          before { allow(RabbitFeed.configuration).to receive(:consumer_exit_after_fail).and_return(true) }
 
-          context 'when the version is lower than 5' do
-            before do
-              module ::Airbrake
-                VERSION = '4.0.0'
+          it 'raises the exception' do
```
Comment (joshuafleck) [inline spec/lib/rabbit_feed/consumer_connection_spec.rb]: Is this description correct?

File: README.md [hunk 2]
```diff
@@ -38,22 +38,23 @@ Sample:
...
+| `connect_timeout`           | The timeout (in seconds) for connecting to the RabbitMQ server                                                                            |
+| `network_recovery_interval` | Recovery interval (in seconds) periodic network recovery will use                                                                         |
+| `auto_delete_queue`         | If true, any queues created will auto-delete upon disconnect                                                                              |
+| `auto_delete_exchange`      | If true, any exchanges created will auto-delete upon disconnect                                                                           |
+| `route_prefix_extension`    | If set, the routing key/queue of the exchange will become env.`route_prefix_extension`.application.event                                  |
+| `consumer_exit_after_fail`  | If set, consumer will stop consuming as soon as it fail to process an event. By Default, it will continue to consume and log the error(s) |
```
Comment (joshuafleck) [inline README.md]: `it fail` -> `it fails`

Comment (joshuafleck) [inline README.md]: `By Default` -> `By default`

File: spec/lib/rabbit_feed/consumer_connection_spec.rb (hunk 1 above)
Comment (PeterWuMC) [inline spec/lib/rabbit_feed/consumer_connection_spec.rb]: damn, i kept reminding myself to change it... and at the end forgot...

**Comment by joshuafleck** [comment]
Nice! :+1: 

signed-off-by: @joshuafleck

**Pull request merged**